src/main/java/de/uos/informatik/ko/rcp/geneticAlgo/GeneratePop.java

```diff
@@ -0,0 +1,101 @@
+package de.uos.informatik.ko.rcp.geneticAlgo;
+
+import java.lang.reflect.Array;
+import java.lang.Integer;
+import java.util.*;
+import de.uos.informatik.ko.rcp.Utils;
+import de.uos.informatik.ko.rcp.Instance;
+
+
+public class GeneratePop {
+
+    private GeneratePop() {}
+    //change to Hashset of ArrayLists
+    public static HashSet<ArrayList<Integer>> generatePop(Instance instance, Integer pop_size,Random myGenerator){
+        // all activities without predecessor are possible candidates
+        HashSet<ArrayList<Integer>> results = new HashSet<ArrayList<Integer>>();
+        HashMap<Integer,HashSet<Integer>> mypredecessors;
+        HashSet<Integer> candidates = new HashSet<Integer>();
+        //scheduled activities
+        ArrayList<Integer> scheduled = new ArrayList<Integer>();
+        for (int i=0; i< pop_size*5; i++) {
+            mypredecessors = Utils.buildPredecessorMap(instance);
```

Comment:
Does this need to be rebuilt on every iteration?

Reply:
Because of line 46 in the code. I remove elements from the predecessor map. I need a "fresh" version of the predeccessor map in every iteration. But your point is valid, since i think it would be more efficient if we create the complete predecessor map only once and copy it in every new iteration.

Reply:
Yeah, I think a copy will be faster for sure :+1: